### ISSUE TITLE
SalesforceAutoField does not inherit from fields.AutoField

### ISSUE THREAD

**Issue by lucaswiman** [django-salesforce/django-salesforce#90]
I'm running into an error running tests on a postgres (non-Salesforce) database. The issue is that django doesn't respect `SalesforceAutoField.get_internal_type`, and generates errors like the following:
```
IntegrityError: null value in column "Id" violates not-null constraint
```

Django is not properly viewing the field as an auto-increment primary key. I think this is caused by a bug in Django [here](https://github.com/django/django/blob/01ab84c61330ffa5ac87c637249611c5e5343e57/django/db/models/base.py#L800), where it checks whether the field is an instance of `AutoField` instead of checking that `f.get_internal_type() == 'AutoField'`. The Django source code seems to be inconsistent on this point, sometimes checking `get_internal_type`, and sometimes checking whether the field is an instance of `AutoField`. I'm going to submit a patch with test cases to Django, but an easy fix in `django-salesforce` itself would be to have `SalesforceAutofield` inherit from `fields.AutoField`.

## How to reproduce
Set your settings file to use a postgres db:
```python
DATABASES = {
    'default': {
        'NAME': 'something',
        'ENGINE': 'django.db.backends.postgresql_psycopg2',
        'USER': 'django',
        'PASSWORD': 'secret',
    },
}
```
Define a simple model:
```python
from salesforce import models as sf_models
from django.db import models

class Foo(sf_models.Model):
    class Meta:
        managed=True
```

Run:
 ```
$ DJANGO_SETTINGS_MODULE=django_salesforce_example python manage.py syncdb
$ DJANGO_SETTINGS_MODULE=django_salesforce_example python manage.py shell
```
Then run in the python shell run
```python
>>> from django_salesforce_example.models import *
>>> Foo.objects.create()
```
This generates the following stack trace:
```pytb
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/Users/lucaswiman/scratchpads/work/django_salesforce_example/.venv/default/lib/python2.7/site-packages/django/db/models/manager.py", line 157, in create
    return self.get_queryset().create(**kwargs)
  File "/Users/lucaswiman/scratchpads/work/django_salesforce_example/.venv/default/lib/python2.7/site-packages/django/db/models/query.py", line 322, in create
    obj.save(force_insert=True, using=self.db)
  File "/Users/lucaswiman/scratchpads/work/django_salesforce_example/.venv/default/lib/python2.7/site-packages/django/db/models/base.py", line 545, in save
    force_update=force_update, update_fields=update_fields)
  File "/Users/lucaswiman/scratchpads/work/django_salesforce_example/.venv/default/lib/python2.7/site-packages/django/db/models/base.py", line 573, in save_base
    updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)
  File "/Users/lucaswiman/scratchpads/work/django_salesforce_example/.venv/default/lib/python2.7/site-packages/django/db/models/base.py", line 654, in _save_table
    result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)
  File "/Users/lucaswiman/scratchpads/work/django_salesforce_example/.venv/default/lib/python2.7/site-packages/django/db/models/base.py", line 687, in _do_insert
    using=using, raw=raw)
  File "/Users/lucaswiman/scratchpads/work/django_salesforce_example/.venv/default/lib/python2.7/site-packages/django/db/models/manager.py", line 232, in _insert
    return insert_query(self.model, objs, fields, **kwargs)
  File "/Users/lucaswiman/scratchpads/work/django_salesforce_example/.venv/default/lib/python2.7/site-packages/django/db/models/query.py", line 1514, in insert_query
    return query.get_compiler(using=using).execute_sql(return_id)
  File "/Users/lucaswiman/scratchpads/work/django_salesforce_example/.venv/default/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 903, in execute_sql
    cursor.execute(sql, params)
  File "/Users/lucaswiman/scratchpads/work/django_salesforce_example/.venv/default/lib/python2.7/site-packages/django/db/backends/util.py", line 69, in execute
    return super(CursorDebugWrapper, self).execute(sql, params)
  File "/Users/lucaswiman/scratchpads/work/django_salesforce_example/.venv/default/lib/python2.7/site-packages/django/db/backends/util.py", line 53, in execute
    return self.cursor.execute(sql, params)
  File "/Users/lucaswiman/scratchpads/work/django_salesforce_example/.venv/default/lib/python2.7/site-packages/django/db/utils.py", line 99, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/Users/lucaswiman/scratchpads/work/django_salesforce_example/.venv/default/lib/python2.7/site-packages/django/db/backends/util.py", line 53, in execute
    return self.cursor.execute(sql, params)
IntegrityError: null value in column "Id" violates not-null constraint
DETAIL:  Failing row contains (null).
```

**Comment by philchristensen** [comment]
Whoops, sorry about the accidental close, "fat-fingered" it.

**Comment by hynekcer** [comment]
Thanks for this issue. I fixed it on our side. Even if they never fix it, they will probably remember that someone can use a customized AutoField and they don't break it like a shot.
  
Our fix opens the way to do django-salesforce more and more compatible with Django if it the current solution will be impassable in future Django versions, but currently is the SalesforceModel the most practical. (I got an idea of providing a conversion method of primary key to a 90-bit integer and accepting an integer between 2**72 and 2**90 as a primary key converted to the proper ID string. I have registered a big problem with custom django-salesforce "_default_manager" and Django 1.8 alpha, but it was still viable finally with the current solution.)
